pkg/templateservicebroker/servicebroker/deprovision.go

```diff
@@ -55,9 +57,35 @@ func (b *Broker) Deprovision(u user.Info, instanceID string) *api.Response {
 		return api.Forbidden(err)
 	}
 
-	opts := metav1.NewPreconditionDeleteOptions(string(brokerTemplateInstance.UID))
+	templateInstance := brokerTemplateInstance.Spec.TemplateInstance
+	opts := metav1.NewPreconditionDeleteOptions(string(templateInstance.UID))
 	policy := metav1.DeletePropagationForeground
 	opts.PropagationPolicy = &policy
+	err = b.templateclient.TemplateInstances(templateInstance.Namespace).Delete(templateInstance.Name, opts)
+	if err != nil {
+		if !kerrors.IsNotFound(err) {
+			return api.InternalServerError(err)
+		}
+	}
+	// Wait for the template instance to go away due to foreground GC propagation
+	err = wait.Poll(time.Second, 30*time.Second, func() (bool, error) {
+		v, err := b.templateclient.TemplateInstances(templateInstance.Namespace).Get(templateInstance.Name, metav1.GetOptions{})
+		if err == nil && v == nil {
```

Comment:
`if err != nil && kerrors.IsNotFound(err) {` ?

Reply:
yeah i guess it should have been. luckily this disaster of a PR is going away :)